packages/time/src/utils/endOf.ts

```diff
@@ -1,67 +1,68 @@
 import type { Temporal } from '@js-temporal/polyfill'
 
+type ViewUnit = 'month' | 'week' | 'day' | 'workWeek' | 'decade' | 'year'
+
+interface EndOfParams {
+  date: Temporal.ZonedDateTime
+  unit: ViewUnit
+  viewModeValue?: number
+  firstDayOfWeek?: number
+}
+
 /**
  * Helper function to get the end of a given temporal unit.
- * @param date {Temporal.ZonedDateTime} - The date for which the end of the unit is needed.
- * @param unit {Unit} - The unit for which to find the end ('day', 'week', 'month', 'year', 'workWeek', 'decade').
+ * @param {EndOfParams} params - The parameters for the endOf function.
  * @returns {Temporal.ZonedDateTime} The end of the given unit.
  */
-export const endOf = (date: Temporal.ZonedDateTime, unit: 'day' | 'week' | 'month' | 'year' | 'workWeek' | 'decade'): Temporal.ZonedDateTime => {
+export function endOf({
+  date,
```

Comment:
Like the direction here. I started a discussion in the time-maintainers discord around these types of methods...